src/UI/Implementation/Component/Signal.php

```diff
@@ -15,12 +16,50 @@ class Signal implements \ILIAS\UI\Component\Signal {
 	protected $id;
 
 	/**
+	 * @var array
+	 */
+	protected $options = array();
+
+	/**
 	 * @param string $id
 	 */
 	public function __construct($id) {
 		$this->id = $id;
 	}
 
+	/**
+	 * @inheritdoc
+	 */
+	public function getId() {
+		return $this->id;
+	}
+
+	/**
+	 * @return array
+	 */
+	public function getOptions() {
+		return $this->options;
+	}
+
+	/**
+	 * @param string $key
+	 * @param mixed $value
+	 */
+	protected function addOption($key, $value) {
+		$this->options[$key] = $value;
+	}
+
+	/**
+	 * @param string $key
+	 * @return mixed|null
+	 */
+	protected function getOption($key) {
+		return (isset($this->options[$key])) ? $this->options[$key] : null;
+	}
```

Comment:
When reading this PR for the first time, I thought that this `addOption` and `getOption` would be a public API, which I would not have liked. But now I see that they are an implementation detail, right? The Question was: If you use `addOption("someOption", "some value")` somewhere: Why don't you just have an `addSomeOption`-method on the signal you want to add that option to? But the question is a lot less interesting when this is an implementation detail...